recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,238 @@
+        "fPIC": [True, False],
+        "shared": [True, False],
+        "historize": [True, False],
+        "logging_level": ["Fatal", "Error", "Warning", "Info", "Debug", "Trace"],
+        "subscription": [True, False],
+        "subscription_events": [True, False],
+        "methods": [True, False],
+        "dynamic_nodes": [True, False],
+        "single_header": [True, False],
+        "multithreading": [True, False],
+        "imutable_nodes": [True, False],
+        "web_socket": [True, False],
+        "discovery": [True, False],
+        "discovery_semaphore": [True, False],
+        "discovery_multicast": [True, False],
+        "query": [True, False],
+        "encription": [True, False],
+        "json_support": [True, False],
+        "pub_sub": ["None", "Simple", "Ethernet", "Ethernet_XDP"],
+        "data_access": [True, False],
+        "compiled_nodeset_descriptions": [True, False],
+        "namescpae_zero": ["MINIMAL", "REDUCED", "FULL"],
+        "embedded_profile": [True, False],
+        "typenames": [True, False],
+        "hardening": [True, False],
+        "cpp_compatible": [True, False],
+        "readable_statuscodes": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "shared": True,
+        "historize": False,
+        "logging_level": "Info",
+        "subscription": True,
+        "subscription_events": False,
+        "methods": True,
+        "dynamic_nodes": True,
+        "single_header": False,
+        "multithreading": False,
+        "imutable_nodes": False,
+        "web_socket": False,
+        "discovery": True,
+        "discovery_semaphore": True,
+        "discovery_multicast": False,
+        "query": False,
+        "encription": False,
+        "json_support": False,
+        "pub_sub": "None",
+        "data_access": True,
+        "compiled_nodeset_descriptions": True,
+        "namescpae_zero": "FULL",
+        "embedded_profile": False,
+        "typenames": True,
+        "hardening": False,
+        "cpp_compatible": False,
+        "readable_statuscodes": True
+    }
+    generators = "cmake", "cmake_find_package"
+
+    def requirements(self):
+        if self.options.encription:
+            self.requires("mbedtls/2.16.3-gpl")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _download_deps(self, zip_link, dep_name):
+        zip_name = dep_name + ".zip"
+        download(zip_link, zip_name)
+        dep_path = self._source_subfolder + "/deps/" + dep_name
+        unzip(zip_name)
+        full_path = str(urlparse(zip_link).path).split("/")
+        dir_name = full_path[2] + "-" + full_path[4]
+        dir_name = dir_name[:-4]
+        files = os.listdir(dir_name)
+        for f in files:
+            shutil.move(os.path.join(dir_name, f), dep_path)
+        os.unlink(zip_name)
+        shutil.rmtree(dir_name)
+
+    def _download_ua_nodes(self):
+        link = ""
+        if self.version == "1.0.0":
```

Comment:
Please use Conan packages for the dependencies.
Or add these extra zips to the conandata.yml file.

Reply:
nodeset is not a c++ dependency, just a submodule. I have the same issue in my PR, where I added a yml for description of submodules by version.

Reply:
This is impossible due to the way github archives projects with sub-module repositories. I opened an issue at open62541 about it, but it seems they can not resolve it.

I could set up separate recipes for the [UA Node set ](https://github.com/OPCFoundation/UA-Nodeset/tree/0777abd1bc407b4dbd79abc515864f8c3ce6812b) and [mdsnd](https://github.com/Pro/mdnsd/tree/4bd993e0fdd06d54c8fd0b8f416cda6a8db18585) repositories, however they would have to be accepted to make this PR work.

Reply:
It seems ua node set might cause packaging problems due to licensing. I will inquirer by open62541 for more info

Reply:
There is already a discussion going on, on how to handle the nodeset submodule.
See: https://github.com/open62541/open62541/issues/3710